tests/functional/DatadirTest.php

```diff
@@ -0,0 +1,437 @@
+                'output' => [
+                    [
+                        'source' => 'invalid_testmetadata',
+                        'destination' => 'out.c-my.invalid_testmetadata',
+                    ],
+                ],
+            ],
+            'parameters' => [
+                'blocks' => [
+                    [
+                        'name' => 'first block',
+                        'codes' => [
+                            [
+                                'name' => 'first code',
+                                'script' => [],
+                            ],
+                        ],
+                    ],
+                ],
+            ],
+        ];
+
+        $this->runAppWithConfig(
+            $config,
+            1,
+            null,
+            "Missing create tables \"invalid_testmetadata\"\n"
+        );
+    }
+
+    public function testMissingAuthorization(): void
+    {
+        $config = [
+            'parameters' => [
+                'blocks' => [
+                    [
+                        'name' => 'first block',
+                        'codes' => [
+                            [
+                                'name' => 'first code',
+                                'script' => [
+                                    'select 1',
+                                ],
+                            ],
+                        ],
+                    ],
+                ],
+            ],
+        ];
+
+        $process = $this->runAppWithConfig(
+            $config,
+            2
+        );
+        $this->assertStringContainsString('Missing authorization for workspace', $process->getErrorOutput());
+    }
+
+    public function testQueryTagging(): void
+    {
+        $createTableQuery = 'create table "query_tag" (
+  "QUERY_TEXT" varchar(200), 
+  "QUERY_TAG" varchar(200)
+);';
+        // phpcs:disable Generic.Files.LineLength
+        $configArray = [
+            'authorization' => $this->getDatabaseConfig(),
+            'parameters' => [
+                'blocks' => [
+                    [
+                        'name' => 'first block',
+                        'codes' => [
+                            [
+                                'name' => 'first code',
+                                'script' => [
+                                    'drop table if exists "query_tag";',
+                                    $createTableQuery,
+                                    sprintf('insert into "query_tag" SELECT QUERY_TEXT, QUERY_TAG FROM TABLE(INFORMATION_SCHEMA.QUERY_HISTORY_BY_SESSION()) WHERE QUERY_TEXT = \'%s\' ORDER BY START_TIME DESC LIMIT 1;', $createTableQuery),
+                                ],
+                            ],
+                        ],
+                    ],
+                ],
+            ],
+        ];
+        // phpcs:enable
+
+        $this->runAppWithConfig($configArray);
+
+        $config = $this->getConfigFromUserConfig($configArray);
```

Comment:
toto je takovy divne zamotany - to nacita config z aplikace vytvorenej behem testu, dal bych tam proste
```
$connection = new Connection($this->getDatabaseConnection())
```
to je uplne primocary

Reply:
upravil jsem to